# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../emacs/ppc-oddity.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

To be able to build and run Emacs on PowerPC we need to wipe some strange
outdated ... assumptions ...

  - Rene Rebe <rene@exactcode.de>

--- emacs-21.3/src/m/macppc.h	2001-10-23 08:43:33.000000000 +0200
+++ emacs-21.3/src/m/macppc.h	2003-09-26 22:26:59.000000000 +0200
@@ -90,26 +90,9 @@
 #define HAVE_TEXT_START
 #endif
 
-/* NAKAJI Hiroyuki <nakaji@tutrp.tut.ac.jp> says this is needed
-   For MkLinux/LinuxPPC.  */
 
-#ifdef LINUX
-#define LINKER $(CC) -nostdlib
-#define LD_SWITCH_MACHINE -Xlinker -m -Xlinker elf32ppc
-/* s/gnu-linux.h defines this to `-z nocombreloc' which does not work here
-   because prefix-args is not used.  */
-#undef LD_SWITCH_SYSTEM_TEMACS
-#define LD_SWITCH_MACHINE_TEMACS -Xlinker -znocombreloc
-#endif
-
-#if 0  /* This breaks things on PPC GNU/Linux ecept for Yellowdog,
-	  even with identical GCC, as, ld.  Let's take it out until we
-	  know what's really going on here.  */
-/* GCC 2.95 and newer on GNU/Linux PPC changed the load address to
-   0x10000000.  */
 #if defined __linux__
 #if __GNUC__ > 2 || (__GNUC__ == 2 && __GNUC_MINOR__ >= 95)
 #define DATA_SEG_BITS  0x10000000
 #endif
-#endif
 #endif /* 0 */
